install.sh

```diff
@@ -114,15 +114,32 @@ systemctl restart systemd-modules-load.service
 [[ -f "${BUILDDIR}/customizations.yaml" ]] && rm -f "${BUILDDIR}/customizations.yaml"
 kubectl get secrets -n loftsman site-init -o jsonpath='{.data.customizations\.yaml}' | base64 -d > "${BUILDDIR}/customizations.yaml"
 
+dist=$(uname | awk '{print tolower($0)}')
+
 # lower cpu request for tds systems (3 workers)
 num_workers=$(kubectl get nodes | grep ncn-w | wc -l)
 if [ $num_workers -le 3 ]; then
-  dist=$(uname | awk '{print tolower($0)}')
   ${ROOTDIR}/shasta-cfg/utils/bin/${dist}/yq m -i --overwrite "${BUILDDIR}/customizations.yaml" "${ROOTDIR}/tds_cpu_requests.yaml"
   kubectl delete secret -n loftsman site-init
   kubectl create secret -n loftsman generic site-init --from-file="${BUILDDIR}/customizations.yaml"
 fi
 
+# inject postgres-backup-s3-credentials for postgres logical backups
+secret=$(kubectl get secret postgres-backup-s3-credentials -n services)
+if [[ ! -z "$secret" ]]; then
+    access_key=$(kubectl get secrets postgres-backup-s3-credentials -ojsonpath='{.data.access_key}' | base64 --decode)
+    secret_key=$(kubectl get secrets postgres-backup-s3-credentials -ojsonpath='{.data.secret_key}' | base64 --decode)
+    ${ROOTDIR}/shasta-cfg/utils/bin/${dist}/yq w -i --style=double "${ROOTDIR}/logical_backup_secrets.yaml" \
+        "spec.kubernetes.services.cray-postgres-operator.postgres-operator.configLogicalBackup.logical_backup_s3_access_key_id" "${access_key}"
+    ${ROOTDIR}/shasta-cfg/utils/bin/${dist}/yq w -i --style=double "${ROOTDIR}/logical_backup_secrets.yaml" \
+        "spec.kubernetes.services.cray-postgres-operator.postgres-operator.configLogicalBackup.logical_backup_s3_secret_access_key" "${secret_key}"
```

Comment:
I think we need to check to see if we should be using sealed secrets to store an encrypted copy of this vs. a plain text version.

It may also be worth exploring if there's already a copy of these in customizations.yaml that we could inject into the chart somehow (I'm not sure if we have a solution for de-duping this kind of config data today).

Reply:
Using a sealed secret would be better, but the code above is basically taking data from one K8S secret and putting it into another.  Also, we're a bit limited now that we're dropping our fork and home-grown logical backup solution, and organizing data for the pure upstream chart.  I think we should move forward with this current change..

Reply:
But doesn't this potentially leave the decoded secrets behind on the system in the customizations.yaml file?

Reply:
I'm not familiar with how those secrets get created to start.   Digging to familiarize myself.

Reply:
The secret postgres-backup-s3-credentials gets created when the storage nodes are bootstrapped and the ceph rgw ansible runs.

Reply:
Do you know if these values above that are decoded (access_key and secret_key) are encrypted?  And if so, what is needed to decrypt them?

Reply:
None of the *s3-credentials secrets created by the ceph rgw ansible are encrypted - they are just put into k8s secrets base64 encoded

Reply:
For the install case - If I add to  https://github.com/Cray-HPE/csm/blob/main/vendor/stash.us.cray.com/scm/shasta-cfg/stable/customizations.yaml
```
spec:
  kubernetes:
    services:
      cray-postgres-operaor:
        sealedSecrets:
          - '{{ kubernetes.sealed_secrets.postgres-backup-s3-credentials | toYaml }}'
```
And update the cray-postgres-operator to include
* template/sealedsecrets.yaml
* values.yaml
```
# A list of sealedSecrets passed in to be deployed.
sealedSecrets: []
```
Then I believe the sealedsecret will be encrypted and the secret decrypted/unsealed by the controller .
With those changes, would it be OK to inject the decoded secrets into customizations.yaml as I've already done in this PR or it that still exposed?

Reply:
Yes, that should work but you'll need to make sure to run the customizations.yaml through and encrypt process to get it encrypted before you pushed the site-init secret back.

https://github.com/Cray-HPE/docs-csm/blame/main/install/prepare_site_init.md#L425-L429

Reply:
The sealed secret in customiations.yaml should look similar this before running it through the encrypt (using the "generate" field.
```
    sealed_secrets:
      cray-keycloak:
        generate:
          name: keycloak-certs
          data:
            - type: static_b64
              args:
                name: certs.jks
                value: /u3+7Qbbbbbbbbbb5yXvSDt11bGXyBA9M2iy0/5i1Tg=
                ```
                
Then after the encrypt (and before it gets pushed to k8s) it will look similar to this:
```
    sealed_secrets:
      cray-keycloak:
        kind: SealedSecret
        apiVersion: bitnami.com/v1alpha1
        metadata:
          name: keycloak-certs
          namespace: services
          creationTimestamp: null
          annotations:
            sealedsecrets.bitnami.com/cluster-wide: "true"
        spec:
          template:
            metadata:
              name: keycloak-certs
              namespace: services
              creationTimestamp: null
              annotations:
                sealedsecrets.bitnami.com/cluster-wide: "true"
          encryptedData:
            certs.jks: AgBvYdZVw6GHOwyaDAmh/VrHrfAFCQlBSPW3GV5z/Pjj19ElyY08NerJGzcZwRwx9rqqforo4F/9W+mKibT/0wY0VfGbpRAOKcDEu8W48G9cGijjWYy/hPGKr5Zlj9rjynJyHvOPHyEwxZhUgSvAOKrrCkOal+vcYlHjlby5+ozjIVQ1WTgGct04muKuE1DzoRcKKJVIrd513DXyvlmtAX/2ar6xr3Pz/bnhMS+iWe23odboc78wxuLoD1x38E7lBod03iAHih+ZwtIcEMoZV0b8oRarbpm5dV2BIPnwwYrOdcpxJrfizFMDms+0H/TXxoYHPDRbUwj47zZVjLFuF7fCyJlDbMfSeKpbs/PR/FrKQ/Rtz5Ex79CDgGIdvd0kpuqlMsoYOJDDhzqGedMH83uugZlCqh9Z9NsOq7yzk5SwVRtkGwdPWUyWLRet62AGbBUwajtBTxZJPUj3/gu4G09bn8ePbcFYViDPGv8aQRjSa3HhDfm1c9Xh6+5t/ucrFfBkYEZEmTEPiXTJz+uVqXTRpjYQJN2fYKXmCQvItiDrY4rrLHb6zLMCalucBfiQIx25d5C9PkLdCixQ0D5vuZnK5qVrM3/oVyIxNlMVYBxvoD60lkNHYtl+ZikNREQUnvdMbkMGNbGE8ZLKI1vrt2VZX6p8UZMqh2uzsAruI0NcXfFlSpqXx9iMWK4eeKGRshQqNYVE5hDSW63OUIiACU7dwA2vWUkBI0vPja3HLsthnlEFZ0V7H/EgIVpiTctmnHtgDB0u5xiGSbFVysrCg0QWw+3zKUREtHvJ6MmDCUy/e/LdS05YYYbrL0Z8AUi/HNNjA2iMid/BvS83icL3/I4XkeXYPp/jXFRDWitJ6DXZdZYN1+Qu/Qo5Xu6z3L5n/e5H4bGXyMdlYNswQ3ja4PTSnIPIHuxanNcGgSCSBBf/Y2HaJrNLzZvRsZQ7cAYHjJWsMCWJBbgGyFfsmCbcUvgPtBwSvicxiBXf3hJCYQ0aenhDbG1ZXNFeJwr05kiRQnpgf8jB231i/uDIGTuy9IuvCWIZGfYK1A23kYI/Q+9JUbAI4X8New3BCsB7UPIIiH6BaXrroN0e/K3w2hn9BnrTNBDmsD28fvgmEiGZ4XLDzrgTTBIZhveyzhFz2/A7npSweRcFnVXbSxrLWF3bzHhloO49MLL8jYrt0E/t01SiZFIabrz2g/S9yCQgegq/Q0zfgxmb+xPhHBKZ2NYlF0nlAtdh4I4/C6guI32m0flVjPU96J2nHaNRDi/1Y2U9Af5P3WjEiNDQsZfFGZ818r3rs4+KZKkV7+hgJEEEjvnTnx+CVjrM/c9WQQvQ3uCUIY5WL0YUtDM0fIQlsK9girS/Te12N1xld7wAnOE800APGXT0JZ4lFT5SKc7mjFhiM54MOYoCfLcty/wZ7CvTsEz8AXBc8bDF4AuUDijgW5nzC5RA0bACjgi1Ol0PrenwknBGq2exOAOY6pIp1j8Vqw+WSREBS+MLElM02InD95SGzrRluw9fmsGPBC9HvLWRVtFm8tYwW6XRW2yjd2fAyVMBVDt7jV+8NgByc6axm29PJSNgDjUcpLST/4Wsh19Zom0+nXFuneTzaCOS4u4ceXs8lJYCqYLbiGhpRUMzjBR+i+y6tLTFQNXa21Pkor8wUXjtR9AJFQAx01eBf1xK3f0vyJbDhSXUiC36R2F8627p2EYUm6mRbOvp49QWt3RuXmdzLuDzFn/KFdI5lx/KI+xL5bI1tuoSXzyYrihQthigCPPonTzlXuEk4Pbu2In5YD++kJ86kymHsdX829ZZwnLDVaniIuA0/LCnByf45GZnF2Fu7b24oAoXLBoV+MeCVBt7Q1JMDkB1wmKsqgtm32tMvOpAwZgQU/cumJgE0GlktZ7jCxM5UCQW0UfeC8QB1UUV0So1zHZz+//ykPZtNArAKgcwP1dhMJtVpdaDrUDW/u6/bXUex3NkCr6omyhPCjAcZzenmyAM3qbIRH6fKy2EiahFHiWKXF6jnjrPcHJjqbxhwzBOpOmlPjP5j+4lpHqcIpd5eemG+SqMrT6WaF+q3qR022xSfAA3yRRrFf3l/qAHsQlaWW3QWmyEH9NyLAtFfPFrqoIHXJCMK5iVEI30SAlF3T7wX98ST+fV+ExcZf8rFX9hwmijEA23F1qcGSoHb0KZa2QNB1EiLajt3NmBVLW4OKW8RfiBR/03SXk8tD+Ae62hPbcxKg5kew5x3YFYSEZQHW/crzb7beS60sNMlgFJpxmCDhOFsPx1t2vthVfe6+YAI1tQDVtKx58WhBUtLhPZvjMnpDquvywp1IfSmin5v/MkkeQpWv6IZzsY7ZI2YbE4PIzwtmiWv+18LYroR/Xeh0qhrnZyioNqwU/S7rFBYCxHBEaXBgQ4yrIESzt5qWQYhdBeYyeNmihXAjI3wZpCA2Cah0YnMDfUcpsWsfXUHgTTAXhVBS8kQR42PPL6z8u/Ip/bfDVNGnNZfeGYpNLmaPbI
            ```

Reply:
Updated above comment with a better example that probably more closely matches what you might need.   I think you would want "type: static" since your secret is already generated.